iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -250,7 +263,7 @@ T& vector<T, Capacity>::front() noexcept
 }
 
 template <typename T, uint64_t Capacity>
-const T& vector<T, Capacity>::front() const noexcept
+inline typename vector<T, Capacity>::const_reference vector<T, Capacity>::front() const noexcept
```

Comment:
Why is this better? I would leave as it is expect the missing inline, this should be added. This goes for all aliases except `value_type` which is an API element for developer to get the underlying type of a vector.

Reply:
Typedefs like `const_reference` or `const_iterator` are part of std::vector. To use them was also proposed by @elBoberido (see https://github.com/eclipse-iceoryx/iceoryx/issues/488#issuecomment-760158500). 
Do you propose to remove them from both, the vector.hpp and the vector.inl? Or keep them at vector.hpp but remove them at vector.inl?
I still think that keeping vector.hpp and vector.inl method signatures in sync with each other and also with stl::vector is the best approach.